common/src/main/java/com/higgs/common/handler/extension/ExtensionHandler.java

```diff
@@ -65,15 +72,18 @@ Map<String, Object> copyFieldTemplateValues(@NonNull final HandlerDefinition pro
     Map<String, Object> copyFieldTemplateValues(@NonNull final Map<String, Object> prototypeDef, @NonNull final Map<String, Object> extensionDef) {
         // for every expected key in the extended handler's prototype definition, get that value from the extension's
         // handler def and use it as input
-        return prototypeDef.keySet().stream()
-                .filter(extensionDef.keySet()::contains)
+        return extensionDef.keySet().stream()
+                .filter(prototypeDef.keySet()::contains)
                 .collect(Collectors.toMap(key -> key, extensionDef::get));
     }
 
     Map<String, Object> interpolateFieldValues(@NonNull final Map<String, Object> toInterpolate, @NonNull final ExtensionHandlerRequest request) {
         return toInterpolate.entrySet().stream()
-                .filter(entry -> entry.getValue() instanceof String)
-                .peek(strEntry -> strEntry.setValue(this.getJinjaEngine().render(((String) strEntry.getValue()), request)))
+                .peek(entry -> {

```

Comment:
try not to use peek

Reply:
peek is only a problem when using a terminal operation that does not take it into account (collect is fine in this case)